invoice/src/main/java/org/killbill/billing/invoice/api/user/DefaultInvoiceUserApi.java

```diff
@@ -718,14 +734,35 @@ private VersionedCatalog getCatalogSafelyForPrettyNames(final InternalTenantCont
 
     @Override
     public void voidInvoice(final UUID invoiceId, final CallContext context) throws InvoiceApiException {
-        final InternalCallContext internalCallContext = internalCallContextFactory.createInternalCallContext(invoiceId, ObjectType.INVOICE, context);
-        Invoice invoice = getInvoice(invoiceId, context);
+        final WithAccountLock withAccountLock = new WithAccountLock() {
+            @Override
+            public Iterable<DefaultInvoice> prepareInvoices() throws InvoiceApiException {
+                final InternalCallContext internalCallContext = internalCallContextFactory.createInternalCallContext(invoiceId, ObjectType.INVOICE, context);
 
-        if (invoice.getNumberOfPayments() > 0) {
-            canInvoiceBeVoided(invoice);
-        }
+                final InvoiceModelDao rawInvoice = dao.getById(invoiceId, internalCallContext);
+                if (rawInvoice.getIsRepaired()) {
+                    // TODO ErrorCode for InvoiceApiException
```

Comment:
Add a task for 0.24?

Reply:
I created a [generic task](https://github.com/killbill/killbill/issues/1501) to do this.